### DISCUSSION TITLE
TabSessionManager - Backup is 4.5GB

### DISCUSSION THREAD

**Discussion by ginamarina101** [sienori/Tab-Session-Manager#672] [Q&A] [answered]
I was trying to locate where the TabSessionManager files are, and I noticed that the "TabSessionManager - Backup" folder in the Downloads folder is 4.5GB.  There are 52 json files saved in the folder. What on earth is it saving? Aren't these just text files?

**Comment by Newtomic** [comment]
Yes that's true those are "just text files"...
The thing is that in those text files (*.json) there could be anything you could imagine related to those sessions (eg: favicons, state of each tab/window, etc.).
I have about 170 sessions total counting with the autosave sessions (I defined to have "only" 50 of each type), I did a simple backup/export and is about ~12MB, so if you have hundreds or more sessions that will add-up, and a backup or more each day and you get there very fast.

You could do 2 things:
-1. delete old backups.
-2. compress the backups, for example my single ~12MB [12 616 579 bytes] compressed with 7-zip got to only ~209KB [214 128 bytes] (~1.697%), assuming the same compression to your backups, those ~4.5GB could fit in about ~78.198MB.

**Comment by sienori** [comment, marked as answer]
I have implemented incremental backup! https://github.com/sienori/Tab-Session-Manager/commit/72da718c019899906e7de7994195ab18f0e4c545 #710
This allows you to manage backup sessions individually.
Only updated sessions are saved, saving storage.

* One session is saved in one json file.
* When a session is updated, the json file is overwritten.
* Autosave sessions are saved in a separate folder.

![image](https://user-images.githubusercontent.com/29514424/116102215-7c6c5f80-a6e9-11eb-9e51-dd4cc4122c31.png)
![image](https://user-images.githubusercontent.com/29514424/116101851-28617b00-a6e9-11eb-9ead-1d032d5c7723.png)

**Reply by rediffusion** [reply]
@sienori

> I have implemented incremental backup!

Where’s the backup location?

**Reply by sienori** [reply]
The backup is stored in the TabSessionManager - Backup folder under the browser default download.
![image](https://user-images.githubusercontent.com/29514424/167867726-409f8477-3310-40b2-b622-37300835f17d.png)
